backend/src/main/java/com/redhat/cloud/notifications/db/EventResources.java

```diff
@@ -0,0 +1,119 @@
+package com.redhat.cloud.notifications.db;
+
+import com.redhat.cloud.notifications.models.Event;
+import io.smallrye.mutiny.Uni;
+import org.hibernate.reactive.mutiny.Mutiny;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.regex.Matcher;
+
+import static com.redhat.cloud.notifications.routers.EventService.SORT_BY_PATTERN;
+
+@ApplicationScoped
+public class EventResources {
+
+    @Inject
+    Mutiny.Session session;
+
+    @Inject
+    Mutiny.StatelessSession statelessSession;
+
+    // Note: This method uses a stateless session
+    public Uni<Event> create(Event event) {
+        event.prePersist(); // This method must be called manually while using a StatelessSession.
+        return statelessSession.insert(event)
+                .replaceWith(event);
+    }
+
+    public Uni<List<Event>> get(String accountId, Set<UUID> bundleIds, Set<UUID> appIds, String eventTypeName,
+                                LocalDate startDate, LocalDate endDate, Integer limit, Integer offset, String sortBy) {
+        String hql = "SELECT DISTINCT e FROM Event e " +
+                "JOIN FETCH e.eventType et JOIN FETCH et.application a JOIN FETCH a.bundle b " +
+                "LEFT JOIN FETCH e.historyEntries he LEFT JOIN FETCH he.endpoint " +
+                "WHERE e.accountId = :accountId";
+
+        if (bundleIds != null && !bundleIds.isEmpty()) {
+            hql += " AND b.id IN (:bundleIds)";
+        }
+
+        if (appIds != null && !appIds.isEmpty()) {
+            hql += " AND a.id IN (:appIds)";
+        }
+
+        if (eventTypeName != null) {
+            hql += " AND et.name = :eventTypeName";
+        }
+
+        if (startDate != null && endDate != null) {
+            hql += " AND DATE(e.created) BETWEEN :startDate AND :endDate";
+        } else if (startDate != null) {
+            hql += " AND DATE(e.created) >= :startDate";
+        } else if (endDate != null) {
+            hql += " AND DATE(e.created) <= :endDate";
+        }
+
+        if (sortBy == null) {
+            hql += " ORDER BY e.created DESC";
+        } else {
+            Matcher sortByMatcher = SORT_BY_PATTERN.matcher(sortBy);
+            if (sortByMatcher.matches()) {
```

Comment:
I think this should fail the request if doesn't matches.

Reply:
Ah, this is done in the service.

Reply:
Perhaps this method should receive the column and direction as different params (maybe using an enum for the direction here)

Reply:
I didn't put the `sortBy` parsing code in the service because it relies on the `getSortField` method which is tightly linked to the persistence layer. I can change that if you want.

Regarding the enum, I don't think that's necessary since the validating regex will only allow `asc` or `desc`.

Reply:
It's fine.